src/controllers/courses.controller.js

```diff
@@ -27,6 +27,27 @@ function create(req, res, next) {
   });
 }
 
+// Fetch all courses
+function list(req, res, next) {
+  Course.find({}, (err, courses) => {
```

Comment:
you may need to populate the steps here and in the read function

Reply:
Both 'list' and 'read' return the step _ids of the steps for each courses. Should they return the entire step instead?

Reply:
Yes, populate the step fields

Reply:
Okay, I made the revisions in my latest commit "update list and read to populate steps". Let me know if any corrections are needed. Thanks!